src/UserRepository.js

```diff
@@ -1,7 +1,27 @@
 class UserRepository {
-  constructor() {
-
+  constructor(userData) {
+    this.data = userData;
+  }
+  getUserData(id) {
+    let userInfo = this.data.find(user => {
+      return user.id === id;
+    })
+    return userInfo;
+  }
+  getStepGoalAverage() {
+    let stepGoals = [];
+    this.data.forEach(user => {
+      stepGoals.push(user.dailyStepGoal)
+    });
+    let stepGoalAverage = stepGoals.reduce((acc, num) => {
+      acc += num / stepGoals.length;
+      return acc;
+    }, 0);
+    return stepGoalAverage
   }
 }
```

Comment:
I think we might run into problems with this with a larger data set, since it looks like you're dividing by the number of users each time, essentially you're doing (2000/3) + (10000/3) + (6000/3), instead of (2000 + 10000 + 6000)/3
We can avoid that issue by writing it like this:
`getStepGoalAverage() {
return Math.floor(this.data.reduce(((acc, user) => acc + user.dailyStepGoal), 0) / this.data.length);
}`

Reply:
refactored